Add MoveIt Servo Tutorial

tahsinkose opened the pull request:
This PR adds two usage examples of MoveIt Servo with UR5 robot in the new `ur5_moveit_tutorials` package

tahsinkose committed:
Create UR5 MoveIt tutorial package with servo example.

tahsinkose committed:
Run controller spawner nodes only when their arguments are non-empty.

tahsinkose committed:
Use moveit_commander to retrieve active joint names and end-effector.

tahsinkose committed:
Use JointTrajectoryController for servo and MoveIt trajectory execution to work together.

tahsinkose committed:
Move common moveit_servo stuff to separate launch file.

tahsinkose committed:
Create launch file for joint_group_pos_controller based moveit servo.

tahsinkose committed:
Specify only the diff controller.

tahsinkose committed:
Specify joint_group_vel_controller for UR5 to experiment in SIM.

tahsinkose committed:
Remove duplicate params and create launch file for joint_group_vel_controller based MoveIt servo.

tahsinkose commented:
@RobertWilbrandt @fmauch could you please have a look at this PR?

AndyZe commented on ur5_moveit_tutorials/CMakeLists.txt:
```diff
@@ -0,0 +1,18 @@
+
+find_package(catkin REQUIRED)
+
+catkin_package(
+)
+include_directories(
+)
```
```suggestion
catkin_package()
```

AndyZe commented on ur_gazebo/launch/inc/ur_control.launch.xml:
```diff
@@ -53,12 +53,16 @@
-  <!-- Load and start the controllers listed in the 'controllers' arg. -->
-  <node name="ros_control_controller_spawner" pkg="controller_manager" type="spawner"
-    args="$(arg controllers)" output="screen" respawn="false" />
+  <group unless="$(eval arg('controllers') == '')">
```
I don't think there's any reason for `group` if there's just one node in the group? Suggest reverting.

tahsinkose replied:
I should actually have used if clause instead of this 😄

AndyZe reviewed (commented):
It would have been very nice to have this a few years ago. Personally I'd still like to see it merged after the changes get applied. I guess most of the community's efforts go towards ROS2 now, for better or worse...

tahsinkose commented:
@AndyZe addressed your review, PTAL 🙏🏼 

> > It would have been very nice to have this a few years ago. Personally I'd still like to see it merged after the changes get applied. I guess most of the community's efforts go towards ROS2 now, for better or worse...

I can migrate them to ROS 2 when I finally switch to ROS 2 🙈

tahsinkose committed:
Add a launch file that supports joy control.

tahsinkose commented:
@AndyZe friendly ping 🙂

AndyZe commented:
I don't have merge power here

tahsinkose commented:
> I don't have merge power here

Who should we reach then? Do you know that @AndyZe?

tahsinkose commented:
@urfeex could you review this PR and possibly merge?

urfeex commented:
I'm afraid, our resources towards ROS 1 are rather limited at the moment.
